packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -428,6 +429,9 @@ class WebAssetServer implements AssetReader {
         ),
         appMetadata: AppMetadata(hostname: hostname),
       ),
+      // Defaults to 'chrome' if deviceManager or specifiedDeviceId is null,
+      // ensuring the condition is true by default.
+      enableDebuggingSupport: (globals.deviceManager?.specifiedDeviceId ?? 'chrome') == 'chrome',
```

Comment:
IIUC from the linked change (https://github.com/dart-lang/webdev/pull/2601) this will inject the code required to support debugging into the page but that doesn't mean debugging actually works?

Does `globals.deviceManager?.specifiedDeviceId` evaluate to `null` when someone runs `flutter run -d web-server`? It isn't clear to me why we want to default it to be true when we don't know if it will work or not.